copyspecial.py

```diff
@@ -19,23 +18,82 @@
 # +++your code here+++
 # Write functions and modify main() to call them
 
-def main():
-    # This snippet will help you get started with the argparse module.
+
+def get_special_paths(dir):
+    path = os.path.abspath(dir)
+    files_list = os.listdir(path)
+    special_paths = []
+    # instructions say in all the directories, but there is only one directory
+    # currently do you mean do the os.path.walk?, or do you mean in the
+    #  directory i indicate??
+    for file in files_list:
+        if re.search(r'__(\w+)__', file):
+            special_paths.append(os.path.abspath(file))
+    return special_paths
+
+
+def copy_paths(paths_to_copy, dest_dir):
+    if dest_dir[0] != '/':
+        dest_dir = '/' + dest_dir
+    dest_dir = os.getcwd() + dest_dir
+    if not os.path.isdir(dest_dir):
+        os.mkdir(dest_dir)
+    for file in paths_to_copy:
+        shutil.copy(file, dest_dir)
+
+
+def zippy(paths, zippath):
+    a = os.path.abspath(zippath)
+    for file in paths:
+        cmd = "zip -j " + zippath + " " + file
+        (status, _) = commands.getstatusoutput(cmd)
+        if not os.path.exists(os.path.dirname(a)):
+            print("oh no!")
+            print("zip error: no file at: " + zippath)
+            print('failed command: ' + cmd)
+            return
+        if status != 0:
+            print("oh no!")
+            print("zip error: oh no - couldn't do it!")
+            print('failed command: ' + cmd)
+            return
+        print(cmd)
+
+
+def create_parser():
     parser = argparse.ArgumentParser()
-    parser.add_argument('--todir', help='dest dir for special files')
-    parser.add_argument('--tozip', help='dest zipfile for special files')
-    # TODO need an argument to pick up 'from_dir'
+    parser.add_argument(
+        '--todir', help='copy files to given directory', action='store_true')
+    parser.add_argument(
+        '--tozip', help='zips files into folder', action='store_true')
+    parser.add_argument('dir', help='directory assigned', nargs='+')
```

Comment:
you should remove the nargs argument. This is setting the argument to a list, unnecessarily. You're only accepting one directory to compress. You've set the arg to accept multiple values i.e. directories, but they index access is used to access the 1st position of args.dir as args.dir[0]. Remove nargs and you can access this as args.dir instead.